listener/templates.go

```diff
@@ -61,14 +61,45 @@ func TemplateDelete(template mqueue.TemplateResponse) error {
 	tStart := time.Now()
 	defer utils.ObserveSecondsSince(tStart, templateMsgHandlingDuration.WithLabelValues(TemplateEventDelete))
 
-	err := database.DB.
-		Delete(&models.Template{}, "uuid = ?::uuid AND rh_account_id = (SELECT id FROM rh_account WHERE org_id = ?)",
-			template.UUID, template.OrgID).Error
-	if err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
-		utils.LogWarn("template", template.UUID, WarnNoRowsModified)
+	// check account
+	accountID, err := middlewares.GetOrCreateAccount(template.OrgID)
+	if err != nil {
+		utils.LogWarn("org_id", template.OrgID, "err", err)
 		return nil
 	}
-	return err
+
+	var templateID int64
+	err = database.DB.Model(&models.Template{}).
+		Select("id").
+		Where("rh_account_id = ? AND uuid = ?::uuid ", accountID, template.UUID).
+		// use Find() not First() otherwise it returns error "no rows found" if uuid is not present
+		Find(&templateID).Error
+	if err != nil {
+		utils.LogWarn("template", template.UUID, "err", err)
+		return nil
+	}
+
+	// unassign systems from the template
+	err = database.DB.Model(&models.SystemPlatform{}).
+		Where("rh_account_id = ? AND template_id = ?", accountID, templateID).
```

Comment:
when the template is not found then I think it won't be error from the query above but templateID=0 would be added to this query. Could that be a problem? I hope not

Reply:
I added check for` templateID == 0`